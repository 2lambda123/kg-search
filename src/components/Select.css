/*
*   Copyright (c) 2018, EPFL/Human Brain Project PCO
*
*   Licensed under the Apache License, Version 2.0 (the "License");
*   you may not use this file except in compliance with the License.
*   You may obtain a copy of the License at
*
*       http://www.apache.org/licenses/LICENSE-2.0
*
*   Unless required by applicable law or agreed to in writing, software
*   distributed under the License is distributed on an "AS IS" BASIS,
*   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
*   See the License for the specific language governing permissions and
*   limitations under the License.
*/

.kgs-select {
    display: inline-block;
    position: relative;
    width: 100%;
    font-size: var(--font-size-small);
    white-space: nowrap;
}

.kgs-select div {
    display: inline-block;
    margin-bottom: 5px;
    margin-right: 20px;
    color: #555;
    font-size: 22px;
    font-weight: normal;
    line-height: 30px;
}

.kgs-select select {
    min-width: 215px;
    width: calc(100% - 70px);
    background-color: white;
    border-color: white;
    color: var(--color-selected-1);
    font-size: 22px;
    font-weight: normal;
    padding: 9px 50px 9px 25px;
    cursor: pointer;
    -moz-user-select: none;
    -ms-user-select: none;
    -webkit-touch-callout: none;
    -webkit-user-select: none;
    user-select: none;
    -o-transition: box-shadow 0.2s ease-in;
    -webkit-transition: -webkit-box-shadow 0.2s ease-in;
    transition: -webkit-box-shadow 0.2s ease-in;
    transition: box-shadow 0.2s ease-in;
    transition: box-shadow 0.2s ease-in, -webkit-box-shadow 0.2s ease-in;
    -moz-appearance: none;
    -webkit-appearance: none;
    text-transform: capitalize;
}

.kgs-select select:hover,
.kgs-select select:focus {
    -webkit-box-shadow: 3px 3px 6px #cbc9c9;
    box-shadow: 3px 3px 6px #cbc9c9;
    z-index: 1;
}

.kgs-select:after {
    content: "";
    position: absolute;
    top: 50%;
    right: 15px;
    width: 0;
    height: 0;
    margin-top: -3px;
    border-top: 6px solid var(--color-selected-1);
    border-right: 6px solid transparent;
    border-left: 6px solid transparent;
    pointer-events: none;
}

/* Internet Explorer 10 & 11 */
@media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
    .kgs-select select {
        padding: 9px;
    }
    
    .kgs-select:after {
        display: none;
    }
}
