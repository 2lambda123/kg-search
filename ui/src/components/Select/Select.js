/*
*   Copyright (c) 2018, EPFL/Human Brain Project PCO
*
*   Licensed under the Apache License, Version 2.0 (the "License");
*   you may not use this file except in compliance with the License.
*   You may obtain a copy of the License at
*
*       http://www.apache.org/licenses/LICENSE-2.0
*
*   Unless required by applicable law or agreed to in writing, software
*   distributed under the License is distributed on an "AS IS" BASIS,
*   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
*   See the License for the specific language governing permissions and
*   limitations under the License.
*/

import React from "react";
import PropTypes from "prop-types";
import "./Select.css";

export const Select = ({className, label, value, list, onChange}) => {
  if (!Array.isArray(list) || list.length <= 1) {
    return null;
  }
  return (
    <div className={`kgs-select ${className?className:""}`}>
      {label && (
        <div>{label}</div>
      )}
      <select onChange={e => onChange(e.target.value || null)} value={value || ""}>
        {list.map(e => <option value={e.value} key={e.value}>{e.label}</option>)}
      </select>
    </div>
  );
};

Select.propTypes = {
  className: PropTypes.string,
  label: PropTypes.string,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]),
  list:  PropTypes.arrayOf(PropTypes.any),
  onChange: PropTypes.func
};

export default Select;