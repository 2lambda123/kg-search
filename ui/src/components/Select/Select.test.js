/*
 * Copyright 2018 - 2021 Swiss Federal Institute of Technology Lausanne (EPFL)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0.
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * This open source software code was developed in part or in whole in the
 * Human Brain Project, funded from the European Union's Horizon 2020
 * Framework Programme for Research and Innovation under
 * Specific Grant Agreements No. 720270, No. 785907, and No. 945539
 * (Human Brain Project SGA1, SGA2 and SGA3).
 *
 */

import React from "react";
import renderer from "react-test-renderer";
import Enzyme, { mount, shallow, render } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Select from "./Select";

Enzyme.configure({ adapter: new Adapter() });

test("Select component renders initially", () => {
  const component = renderer.create(
    <Select className="className" label="a label" list={[{label: "a label", value: "a value"},{label: "another label", value: "another value"}]} onChange={() => {}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test("Select test className", () => {
  const component = shallow(
    <Select className="className" label="a label" list={[{label: "a label", value: "a value"},{label: "another label", value: "another value"}]} onChange={() => {}} />
  );
  expect(component.hasClass("className")).toBe(true);
});

test("Select test label", () => {
  const component = shallow(
    <Select className="className" label="a label" list={[{label: "a label", value: "a value"},{label: "another label", value: "another value"}]} onChange={() => {}} />
  );
  expect(component.find("div div").text()).toBe("a label");
});

test("Select test number of items", () => {
  const component = render(
    <Select className="className" label="a label" list={[{label: "a label", value: "a value"},{label: "another label", value: "another value"}]} onChange={() => {}} />
  );
  expect(component.find("option").length).toBe(2);
});

test("Select test onchange", () => {
  const fn = jest.fn();
  const component = mount(
    <Select className="className" label="a label" list={[{label: "a label", value: "a value"},{label: "another label", value: "another value"}]} onChange={fn} />
  );
  component.find("select").simulate("change", {target: { value : "another value"}});
  expect(fn.mock.calls.length).toBe(1);
  expect(fn.mock.calls[0][0]).toBe("another value");
});