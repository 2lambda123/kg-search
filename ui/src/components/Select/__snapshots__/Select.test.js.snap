// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component renders initially 1`] = `
<div
  className="kgs-select className"
>
  <div>
    a label
  </div>
  <select
    onChange={[Function]}
    value=""
  >
    <option
      disabled={false}
      value="a value"
    >
      a label
    </option>
    <option
      disabled={false}
      value="another value"
    >
      another label
    </option>
  </select>
</div>
`;
